starboard/build/BUILD.gn

```diff
@@ -0,0 +1,39 @@
+# Copyright 2024 The Cobalt Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import("//build/buildflag_header.gni")
+import("//cobalt/build/configs/cobalt.gni")
+import("//cobalt/build/configs/variables.gni")
+
+declare_args() {
+  use_starboard = target_os != "android"
+  use_starboard_media = is_cobalt
+  use_evergreen = false
+}
+
+# To use these build flags from i.e. "foo.cc", add a dependency from the target
+# containing "foo.cc" on this target, then have "foo.cc" include
+# "starboard/build/starboard_buildflags.h". After that you can use i.e.
```

Comment:
If we use this way, we will need to add `starboard/build/starboard_buildflags.h` to all modified Chromium C++ files, which is not guarded, although this is clean when we define buildflags, it is not guarded when we use it.

On the other hand, if we add these buildflags to build/build_config.h, this will be the only file we modify without guarding it. All other changes, including header files, can be guarded by our buildflags.

Reply:
We could potentially add USE_STARBOARD/IS_STARBOARD to //build/build_config.h, but I don't want to put things in there that do not fit with upstream conventions for what belongs there, as we're trying to add a code in a way that it could eventually be upstreamed. Additionally, having this explicit include makes it clear where we're using cobalt/starboard code.

Reply:
Agreed. In this case, we should clarify the standard ways to add/modify Chromium C++ codes to our Cobalt development doc.